src/main/java/com/maria/library/controller/CategoryController.java

```diff
@@ -0,0 +1,23 @@
+package com.maria.library.controller;
+
+import com.maria.library.facade.ICategoryFacade;
+import com.maria.library.service.ICategoryService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/category")
+public class CategoryController {
+    private final ICategoryFacade categoryFacade;
+
+    public CategoryController(ICategoryFacade categoryFacade) {
```

Comment:
very nice that you can inject interface here. If it really works, I suggest you to change all your code and replace concrete class in constructors with interface.